### ISSUE TITLE
W0612 being falsely raised in specific `raise from` cases

### ISSUE THREAD

**Issue by tushar-deepsource** [PyCQA/pylint#5350]
### Bug description

This is the minimal reproduction, removing anything from this hides the bug:

```python
def func():
    try:
        pass
    except SomeError as e:
        print(e)
    except AnotherError as e:
        raise ValueError from e
```

### Command used

```shell
pylint mytest.py
```

### Pylint output

```shell
mytest.py:6:4: W0612: Unused variable 'e' (unused-variable)
```

### Expected behavior

No error to be raised

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.5
Python 3.10.0 (default, Oct 13 2021, 06:45:00) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

MacOS Monterey

**Comment by tushar-deepsource** [comment]
Okay, just reset my environment variables and tried it again: It doesn't happen anymore.
I'm using a local pylint plugin, somehow that plugin is causing this bug to show up. Will update more about it as I find.

**Comment by tushar-deepsource** [comment]
Okay, it was unrelated. Sorry about that
